pkg/datapath/linux/requirements.go

```diff
@@ -184,17 +185,24 @@ func CheckMinRequirements() {
 				log.Warn("llc version was compiled in debug mode, expect higher latency!")
 			}
 		}
-		// /usr/include/gnu/stubs-32.h is installed by 'glibc-devel.i686' in fedora
-		// /usr/include/sys/cdefs.h is installed by 'libc6-dev-i386' in ubuntu
-		// both files exist on both systems but cdefs.h already exists in fedora
-		// without 'glibc-devel.i686' so we check for 'stubs-32.h first.
-		if _, err := os.Stat("/usr/include/gnu/stubs-32.h"); os.IsNotExist(err) {
-			log.Fatal("linking environment: NOT OK, please make sure you have 'glibc-devel.i686' if you use fedora system or 'libc6-dev-i386' if you use ubuntu system")
-		}
-		if _, err := os.Stat("/usr/include/sys/cdefs.h"); os.IsNotExist(err) {
-			log.Fatal("linking environment: NOT OK, please make sure you have 'libc6-dev-i386' in your ubuntu system")
+		hardwarename, err := loader.GetHardwareName()
+		if err != nil {
+			log.WithError(err).Fatal("Get machine hardware name error!")
```

Comment:
I would be careful with the fatal here. Maybe an error would be good enough? In the end, this whole section is for people that are not running cilium in a container because our container image already has all the dependencies installed.

Reply:
Corrected by https://github.com/cilium/cilium/pull/9900/commits/73ea3c61dac32e5f8c9694f157070a40223a3d4e